proto/palomachain/paloma/evm/user_smart_contract.proto

```diff
@@ -0,0 +1,31 @@
+syntax = "proto3";
+package palomachain.paloma.evm;
+
+import "cosmos_proto/cosmos.proto";
+
+option go_package = "github.com/palomachain/paloma/x/evm/types";
+
+enum DeploymentStatus {
+  DEPLOYMENT_STATUS_UNSPECIFIED = 0;
+  DEPLOYMENT_STATUS_ONGOING = 1;
+  DEPLOYMENT_STATUS_ACTIVE = 2;
+  DEPLOYMENT_STATUS_ERROR = 3;
+}
+
+// UserSmartContract defines user-uploaded smart contracts
+// We keep them in storage, so users can decide to deploy them anywhere
+message UserSmartContract {
+  message Deployment {
+    string chain_reference_id = 1;
+    DeploymentStatus status = 2;
+    string address = 3;
+  }
```

Comment:
Careful storing deployment data as part of the contract. I assume we're using the same architectural approach as we do with the Compass smart contract, or Jobs - meaning we store the data on chain first, then we deploy it as a second step.

Keep in mind we can deploy the contract to the same target chain more than once. We also want to consider the possibility of updating a smart contract. Probably something to consider as an upgrade path in the future.

I suggest to add `created_at` and `updated_at` block heights to this, and make sure consuming logic allows for multiple entries of same `chain_reference_id`. In the future, if we do allow upgrades, they'd have to be versioned as well.

Also, what happens if a user removes the smart contract again? All deployment history will be lost as well. I suppose that's okay, just double checking.

Reply:
I was assuming we only wanted to keep track of the latest deployment for each chain. I'll change it to keep track of all deployments. Including the block height should be enough for that.

> Also, what happens if a user removes the smart contract again? All deployment history will be lost as well.
Yes. It could be confusing to keep track of deployments on contracts we no longer have on chain.